### DISCUSSION TITLE
Writing the device index to disk failed: File or directory 'C:\**\build\node_modules\@zwave-\config\config\devices\index.json' was not included into executable at compilation stage. Please recompile adding it as asset or script.

### DISCUSSION THREAD

**Discussion by Oupsssss** [zwave-js/zwave-js-ui#3967] [General]

**Comment by robertsLando** [comment]
@Oupsssss How did you installed zwave-js-ui?

**Reply by Oupsssss** [reply]
Hi,
Thanks for your answer.
I did it with the standard package available here: https://github.com/zwave-js/zwave-js-ui/releases
I fixed my problem after deleting the directory named "store", and restarted the application.
I may have created this problem by naming my nodes with forbidden characters, such as parentheses.

**Reply by robertsLando** [reply]
@Oupsssss I don't think that's the problem in this case, btw let me know if the issue persist

**Comment by Oupsssss** [comment]
My problem is solved. Thank you for your concern.
Now I bothered by the setting of my set points which suddenly return a value ​​of 22° and back the real set point at the news interview.
I wonder if I should check or uncheck the persistent of HOME ASSISTANT parameters of your application. I can't find a clear explanation for this box (I am a beginner, before I used open Z-Wave.)

![image](https://github.com/user-attachments/assets/1a4233ef-c470-494d-9f47-5568c20bef65)

**Reply by robertsLando** [reply]
Persistent discovery is meant to be used when you have to customize the discovery template for some reason otherwise you can keep it disabled

**Reply by Oupsssss** [reply]
Thank you once.
Domoticz recommends to set them ON: https://www.domoticz.com/wiki/Zwave-JS-UI

![image](https://github.com/user-attachments/assets/81bad8fe-4124-4a9e-970d-a0b2b337f00e)

**Reply by robertsLando** [reply]
Oh yeah sorry another reason why you could want to persist them is if you want to skip the discovery of some entities, in that case you have to make that persistent

**Reply by Oupsssss** [reply]
Thanks for your confirmation. That reassures me.
When the pertistent checkbox was disabled, Domoticz has duplicated some nodes. How can I delete them since they are not marked as dead? I was able to delete them in the domoticz tree via MQTT Explorer, but they are not deleted in the JS-UI node list. Is it possible that they were created into to the stick as well ?

**Reply by robertsLando** [reply]
If you have retained discovery enabled you should delete the retained discovery topics using mqtt explorer, even if I dunno why this happen, maybe for some reason you changed the entity id template in settings?

**Reply by Oupsssss** [reply]
Hello,

Yesterday evening I solved my problem regarding the persistent data, with the trick of your previous message.
Now I need to delete the ghost nodes that were stored into the my stick. Yesterday I did it for one node with the help of the application mamed "ZenSYS-Tools".  (Remark: Why not insert this deleting option into your own application ? Because I seen that  this topic is recurrent on the web.)
Thank you for your support. I appreciate a lot. :)

**Reply by robertsLando** [reply]
@Oupsssss when you say `I need to delete the ghost nodes that were stored into the my stick` what do you mean exactly? Are this nodes actually alive or dead? If dead you could simply call remove failed node against those nodes and they will be removed

**Reply by Oupsssss** [reply]
This is hard to answer to your question, because the **nodes were not dead, but not really alive either**.
I have probably created this situation myself, by lack of knowledge. Let me explain with my rookie's words, if it could helps you :

My old domoticz 2023.1 version was becoming obsolete. So I updated Domotics, and started to migrate from openzwave to Zwave-js-ui with the same Z-Stick.
BUT, not knowing that the nodes were saved into the stick, I had the bad idea of ​​resetting my first two Danfoss setpoints to factory, rather than excluding them from the Z-stick via the usual procedure (even if it was not really necessary to do so). So in-fine, the Z-stick therefore still contained the nodes of the two setpoints, but the 'link" was cut-off.

Afterward, I have included once the 2 setpoints with Zwave-js-ui inclusion procedure without any problem.

BUT! Oddly enough, the two old nodes still returned values ​​that I found in MQTT (Always 22°C for instance). And strange as it may seem, these 22°C values ​​were assigned to the new nodes of the each relevant setpoints, even though they were two different nodes. At least that's what I deduced.
So I had to remove these nodes from the Z-stick to stabilise the setting of my setpoints. 
Unfortuntelly, it is impossible to do it via Zwave-js-ui which seems to only be able to remove the dead nodes.
I finally managed to do it via this procedure :  [https://community.openhab.org/t/remove-a-ghost-z-wave-node-from-habmin/6640/8]

**Reply by robertsLando** [reply]
@AlCalzone What do you suggest to do in this case?

**Reply by AlCalzone** [reply]
Please make a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs) of the actions that don't work. Set it to loglevel `debug` and attach it here as a file (drag & drop into the text field).

**Reply by Oupsssss** [reply]
Thank you both for your concern, BUT I have deleted the two "ghost" nodes with the help of "ZenSYS-Tools". So I can't extract any log anymore. My only suggestion would be to add the function "RemoveFailedNode" function into the "advanced actions" of the your Control Panel. 

_Function Class RemoveFailedNode_
Syntax: RemoveFailedNode(nodeId, successCallback = NULL, failureCallback =
NULL)
Description: Remove failed node from network. Before removing SDK will check that
the device is really unreachable
Parameter nodeId: Node Id to be removed from network
Parameter successCallback: Custom function to be called on function success. NULL
if callback is not needed
Parameter failureCallback: Custom function to be called on function failure. NULL
if callback is not needed

Note: – RemoveFailedNode(): Remove this node as failed. Device should be marked as failed
to remove it with this function.

Source: [Here:](https://platenspeler.github.io/HardwareGuide/Zwave-Gateway/RazBerry/Documentation/zwayDev.pdf)

**Reply by AlCalzone** [reply]
> My only suggestion would be to add the function "RemoveFailedNode" function into the "advanced actions" of the your Control Panel.

This exists! You have to look in the node-specific advanced options instead of the global ones.

**Reply by Oupsssss** [reply]
Do you mean this function ? 
![image](https://github.com/user-attachments/assets/764278c3-5ba2-413a-a2a9-004ef9ee0d00)
If so, it did not work on my case because my nodes were not dead neither failed.
That why I had used  "ZenSYS-Tools" software who allowed me to mark the node as failed status, prior to remove it.

**Reply by AlCalzone** [reply]
No. Open the node dropdown, there is another "advanced" button.

**Reply by Oupsssss** [reply]
I got it !
![image](https://github.com/user-attachments/assets/0b424aac-751d-40f8-8bf8-f1de0dfe849a)
It was so simple! I complicated my life...
We learn something new every day... Thank you for your insight.
